subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileIntegrationTest.groovy

```diff
@@ -583,7 +577,7 @@ class JavaCompileIntegrationTest extends AbstractPluginIntegrationTest {
 
         then:
         executedAndNotSkipped ':fooJar', ':compileJava'
-        outputContains "Malformed jar 'foo.jar' found on classpath."
+        outputContains "Malformed zip 'foo.jar'"
```

Comment:
Nit: I think using `zip` here is slightly misleading for Java developers; it is not required knowledge that JAR files are also ZIP files. Using `archive` instead would be better.

Reply:
Good point.